dask/array/tests/test_atop.py

```diff
@@ -166,7 +224,37 @@
                 (b, "i", {b: (add, _0, _1)}, [(a, "i"), (123, None)]),
                 (c, "j", {c: (add, _0, _1)}, [(b, "j"), (123, None)]),
             ],
-            (c, "j", {b: (add, _1, _0), c: (add, b, _0)}, [(123, None), (a, "j")]),
+            (
+                c,
+                "j",
+                {_unique_dep(b, "j"): (add, _1, _0), c: (add, _unique_dep(b, "j"), _0)},
+                [(123, None), (a, "j")],
+            ),
+        ],
+        # Check case where two distinct indices are used
+        # for the same dependency name
+        # (See: https://github.com/dask/dask/issues/8535)
+        [
+            [
+                (b, "jk", {b: (add, _0, _1)}, [(a, "jk"), (2, None)]),
+                (c, "ijk", {c: (add, _0, _1)}, [(b, "ij"), (b, "jk")]),
```

Comment:
Should you also test repeating the same name-indices combination as an input? Like
```python
(x, "ijk", {c: (add, _0, _1, _2)}, [(b, "ij"), (b, "jk"), (b, "ij")]),
```
or also
```python
(y, "ijk", {c: (sub, _0, _1)}, [(b, "ij"), (b, "jk")]),
```
(so `b_i_j` and `b_j_k` show up as deps multiple times in the list of inputs)

Reply:
I added another test case, but I'm not sure if it covers the case you are thinking of here.